SampleGallery/Samples/SDK 16299/LinearGradients/LinearGradients.xaml.cs

```diff
@@ -0,0 +1,263 @@
+            _changeStopToLightSkyBlue.InsertKeyFrame(1f, _lightSkyBlue);
+            _changeStopToLightSkyBlue.Duration = TimeSpan.FromSeconds(2);
+
+            _changeStopToTeal = _compositor.CreateColorKeyFrameAnimation();
+            _changeStopToTeal.InsertKeyFrame(1f, _teal);
+            _changeStopToTeal.Duration = TimeSpan.FromSeconds(2);
+        }
+
+        // When pointer is pressed whatever is the current set of colors (DeepPink/Honeydew and LightSkyBlue/Teal) is swtiched to the set not displayed
+        private void Pointer_Pressed(object sender, RoutedEventArgs e)
+        {
+            SwitchColorSetOnPressed(_b1GradientStop1, _b1GradientStop2);
+        }
+
+        // That switches the sets of colors (DeepPink/Honeydew and LightSkyBlue/Teal)
+        private void SwitchColorSetOnPressed(CompositionColorGradientStop stop1, CompositionColorGradientStop stop2)
+        {
+            // If stop 1 is currently one of the warmer colors that are set together, it will call animations to switch both stops to the cooler color sets when the mouse clicks on the visuals
+            if (stop1.Color == _deepPink || stop1.Color == _honeydew)
+            {
+                stop1.StartAnimation("Color", _changeStopToLightSkyBlue);
+                stop2.StartAnimation("Color", _changeStopToTeal);
+            }
+            // If stop 1 is currently one of the cooler colors that are set together, it will call animations to switch both stops to the warmer color sets when the mouse clicks on the visuals
+            else if (stop1.Color == _lightSkyBlue || stop1.Color == _teal)
+            {
+                stop1.StartAnimation("Color", _changeStopToDeepPink);
+                stop2.StartAnimation("Color", _changeStopToHoneydew);
+            }
+        }
+
+        // Animates scale of visual that method is called on
+        private void AnimateScale(SpriteVisual target)
+        {
+            target.StartAnimation("Scale", _scaleAnim);
+        }
+
+        // Switch the current colors of the two color stops in the linear gradient brush. Called when first XAML UI Element is entered by the mouse pointer
+        private void ChangeStopColorsOnEntered(CompositionColorGradientStop stop1, CompositionColorGradientStop stop2)
+        {
+            if (stop1.Color == _deepPink)
+            {
+                stop1.StartAnimation("Color", _changeStopToHoneydew);
+                stop2.StartAnimation("Color", _changeStopToDeepPink);
+            } else if (stop1.Color == _teal)
+            {
+                stop1.StartAnimation("Color", _changeStopToLightSkyBlue);
+                stop2.StartAnimation("Color", _changeStopToTeal);
+            }
+        }
+
+        // Switch the current colors of the two color stops in the linear gradient brush. Called when last XAML UI Element is entered by the mouse pointer
+        private void ReverseStopColorsOnEntered(CompositionColorGradientStop stop1, CompositionColorGradientStop stop2)
```

Comment:
It looks like this animation is designe to do nothing if the previous animation hasn't finished. Is that the intention? Might be worth a comment.

Reply:
No, would I had stop animations to the if / else if statements?